zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -702,10 +693,10 @@ private[consumer] object Runloop {
                   restartStreamsOnRebalancing,
                   currentStateRef
                 )
-      _    <- ZIO.logDebug("Starting Runloop").withFinalizer(_ => ZIO.logDebug("Shut down Runloop"))
-      stop <- Ref.make(false)
-      fib  <- runloop.run(stop).forkScoped
-      _    <- ZIO.addFinalizer(ZIO.logTrace("Shutting down Runloop") *> stop.set(true) *> fib.join.orDie)
+      _ <- ZIO.logDebug("Starting Runloop").withFinalizer(_ => ZIO.logDebug("Shut down Runloop"))
+      // Run the entire loop on the blocking thread pool to avoid executor shifts
+      fib <- ZIO.blocking(runloop.run).forkScoped
+      _ <- ZIO.addFinalizer(ZIO.logTrace("Shutting down Runloop") *> commandQueue.offer(StopRunloop) *> fib.join.orDie)
```

Comment:
The suggestion removes double logging.

I was confused about the use of the *> operator. I though: this is not needed in a for comprehension. However, only then I noticed they are all within the finalizer block.
Perhaps some more newlines make this clearer:

```suggestion
      _ <- ZIO.logDebug("Starting Runloop")
      // Run the entire loop on the blocking thread pool to avoid executor shifts
      fib <- ZIO.blocking(runloop.run).forkScoped
      _ <- ZIO.addFinalizer(
         ZIO.logTrace("Shutting down Runloop") *>
         commandQueue.offer(StopRunloop) *>
         fib.join.orDie
      )
```